tower-grpc-interop/src/client.rs

```diff
@@ -278,33 +318,202 @@ impl Testcase {
-                                        "call must be successful",
-                                        result.is_ok(),
-                                        format!("result={:?}", result)
-                                    )
-                            ];
-                            if let Ok(response) = result.map(|r| r.into_inner()) {
-                                assertions.push(test_assert!(
-                                "aggregated payload size must be 74922 bytes",
-                                response.aggregated_payload_size == 74922,
-                                format!("aggregated_payload_size={:?}",
-                                    response.aggregated_payload_size
-                                )));
+                let stream = stream::iter_ok(REQUEST_LENGTHS.iter()
+                    .map(|len| util::client_payload(*len as usize)));
+                core.run(client
+                    .streaming_input_call(Request::new(stream))
+                    .then(|result| {
+                        let mut assertions = vec![
+                                test_assert!(
+                                    "call must be successful",
+                                    result.is_ok(),
+                                    format!("result={:?}", result)
+                                )
+                        ];
+                        if let Ok(response) = result.map(|r| r.into_inner()) {
+                            assertions.push(test_assert!(
+                            "aggregated payload size must be 74922 bytes",
+                            response.aggregated_payload_size == 74922,
+                            format!("aggregated_payload_size={:?}",
+                                response.aggregated_payload_size
+                            )));
+                        }
+                        future::ok::<Vec<TestAssertion>, Box<Error>>(assertions)
+                    }))
+            },
+            Testcase::server_streaming => {
+                use pb::ResponseParameters;
+                let req = pb::StreamingOutputCallRequest {
+                    response_parameters: RESPONSE_LENGTHS.iter().map(|len| {
+                        ResponseParameters::with_size(*len)
+                    }).collect(),
+                    ..Default::default()
+                };
+                let req = Request::new(req);
+                core.run(client
+                    .streaming_output_call(req)
+                    .map_err(|tower_err: tower_grpc::Error<tower_h2::client::Error>| -> Box<Error> {
+                        Box::new(tower_err)
+                    })
+                    .and_then(|response_stream| {
+                        // Convert the stream into a plain Vec
+                        response_stream.into_inner()
+                            .collect()
+                            .map_err(|tower_err: tower_grpc::Error<()>| -> Box<Error> {
```

Comment:
Again, I think this can just be `map_err(Box::new)`

Reply:
Same answer again. It could be made more concise with a helper function but I think a better solution would be to simplify the error types exposed by tower-grpc so that this is not needed in the first place.